### PR TITLE
Process existing ics files from URL

### PR DESCRIPTION
**Pull request by melissawm**
This PR adds functionality to process existing .ics files from a URL and create a compatible calendar. (The goal of this PR is to be able to show the result in scientific-python.org)

Addresses https://github.com/scientific-python/scientific-python.org/issues/246

Feedback is welcome!

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -24,6 +24,7 @@ dependencies = [
     "ics == 0.8.0.dev0",
     "python-dateutil >= 2.8",
     "pyyaml >= 6",
+    "requests",
```
Comment (tupui) [inline pyproject.toml:27]: I would not add an extra dependency to `requests` or another HTTP library as the needs are very limited. [urllib.request](https://docs.python.org/3/library/urllib.request.html) should be more than enough for that.

File: yaml2ics.py [hunk 2]
```diff
@@ -146,7 +147,7 @@ def events_to_calendar(events: list) -> str:
     return cal
 
 
-def files_to_events(files: list) -> (ics.Calendar, str):
+def files_to_events(files: list) -> (ics.Calendar, str, bool):
```
Comment (tupui) [inline yaml2ics.py:150]: Can we add some doc about that somewhere? The other parameters/return values were sort of ok to guess but this is less so to me.

**Review by tupui — COMMENTED** [review]
Thanks Melissa! I only have a suggestion to use stdlib. The rest is maybe some doc and I suppose you were waiting for some feedback before adding a test?

**Comment by melissawm** [comment]
Exactly, just checking if this was reasonable. Thanks for the suggestions - I'll fix them up and add the tests 😁

**Comment by stefanv** [comment]
Given that `yaml2ics` turns yaml into ICS, and these calendars are *already* in ICS format, I wonder if it wouldn't make sense to rather modify the process on the `scientific-python.org` site instead. I.e., have a small Python script that calls `yaml2ics` for files without, say, an `external-ics:` field, otherwise fetch the ICS and put it where it needs to go.

I suggest this, because I think this functionality is unlikely to be required by users of this library, but is obviously already required on scientific-python.org!

**Comment by melissawm** [comment]
That makes sense too - I was wondering what the best approach would be given that we might want to keep track of these calendars in one place (right now this would mean the folder where all yaml files are in scientific-python.org).

**Comment by rkdarst** [comment]
I'm at least somewhat interested in this idea!

It's a good question if there should be something else that merges ICS feeds after yaml2ics does its job.  If someone makes template commands for that, please paste here!

**Comment by stefanv** [comment]
Please see https://github.com/scientific-python/scientific-python.org/pull/519#issuecomment-2324747658

I think, unfortunately, we are not able to provide this functionality.

**Pull request closed without merging**